src/pathwayvis/services/actions/actions.service.ts

```diff
@@ -87,6 +87,35 @@ class UndoKnockout extends Knockout {
   }
 }
 
+@registerAction
+// tslint:disable-next-line
+class SetObjective extends ReactionSetAsObjectiveAction implements Action {
+  public label = 'Set as objective';
+  public type: string = 'reaction:objective:do';
+  public shared: types.MapData;
+
+  public canDisplay(context) {
+    const isRemoved = context.shared.objectiveReaction !== context.element.bigg_id;
+    return context.type === 'map:reaction' && isRemoved;
+  }
+}
+
+@registerAction
+// tslint:disable-next-line
+class UndoSetObjective extends ReactionAction {
+  public label = 'Undo set as objective';
+  public type: string = 'reaction:objective:undo';
+
+  public canDisplay(context) {
```

Comment:
What if we'd turn this into a multi-line logical expression?

    public canDisplay({shared, element}) {
      return shared.objectiveReaction &&
        shared.objectiveReaction === element.bigg_id &&
        context.type === 'map:reaction';
    }